nuve/nuveAPI/cloudHandler.js

```diff
@@ -151,94 +165,125 @@ exports.addNewErizoController = function (msg, callback) {
 exports.keepAlive = function (id, callback) {
     var result;
 
-    if (erizoControllers[id] === undefined) {
-        result = 'whoareyou';
-        log.warn('I received a keepAlive message from an unknown erizoController');
-    } else {
-        erizoControllers[id].keepAlive = 0;
-        result = 'ok';
-        //log.info('KA: ', id);
-    }
-    callback(result);
+    erizoControllerRegistry.getErizoController(id, function (erizoController) {
+
+        if (erizoController === undefined) {
+            result = 'whoareyou';
+            log.warn('I received a keepAlive message from an unknown erizoController');
+        } else {
+            erizoController.keepAlive = 0;
+            erizoControllerRegistry.updateErizoController(id, erizoController);
+            result = 'ok';
+            //log.info('KA: ', id);
+        }
+        callback(result);
+    });
 };
 
 exports.setInfo = function (params) {
     log.info('Received info ', params, '. Recalculating erizoControllers priority');
-    erizoControllers[params.id].state = params.state;
-    recalculatePriority();
+    erizoControllerRegistry.getErizoController(params.id, function (erizoController) {
+        erizoController.state = params.state;
+        erizoControllerRegistry.updateErizoController(params.id, erizoController);
+    });
 };
 
 exports.killMe = function (ip) {
     log.info('[CLOUD HANDLER]: ErizoController in host ', ip, 'wants to be killed.');
 };
 
-exports.getErizoControllerForRoom = function (room, callback) {
+var getErizoControllerForRoom = exports.getErizoControllerForRoom = function (room, callback) {
     var roomId = room._id;
 
-    if (rooms[roomId] !== undefined) {
-        callback(erizoControllers[rooms[roomId]]);
-        return;
-    }
-
-    var id,
-        attempts = 0,
-        intervarId = setInterval(function () {
-
-        if (getErizoController) {
-            id = getErizoController(room, erizoControllers, ecQueue);
-        } else {
-            id = ecQueue[0];
+    roomRegistry.getRoom(roomId, function (room) {
+        var id = room.erizoControllerId;
+        if (id) {
+            erizoControllerRegistry.getErizoController(id, function (erizoController) {
+                if (erizoController) {
+                    callback(erizoController);
+                } else {
+                    room.erizoControllerId = undefined;
+                    roomRegistry.updateRoom(room._id, room);
+                    getErizoControllerForRoom(room, callback);
+                }
+            });
+            return;
         }
 
-        if (id !== undefined) {
+        var attempts = 0;
+        var intervarId;
```

Comment:
ok ok... it seems it [doesn't exist](http://www.wordreference.com/definition/intervar) in English. Fixed!